packages/core/src/render3/di.ts

```diff
@@ -344,6 +347,120 @@ export function injectAttributeImpl(tNode: TNode, attrNameToInject: string): str
   return null;
 }
 
+/**
+ * Returns a `TNode` of the location where the current `LView` is declared at.
+ *
+ * @param lView an `LView` that we want to find parent `TNode` for.
+ */
+function getDeclarationTNode(lView: LView): TNode|null {
+  const tView = lView[TVIEW];
+
+  // Return the declaration parent for embedded views
+  if (tView.type === TViewType.Embedded) {
+    ngDevMode && assertDefined(tView.declTNode, 'Embedded TNodes should have declaration parents.');
+    return tView.declTNode;
+  }
+
+  // Components don't have `TView.declTNode` because each instance of component could be
+  // inserted in different location, hence `TView.declTNode` is meaningless.
+  // Falling back to `T_HOST` in case we cross component boundary.
+  if (tView.type === TViewType.Component) {
+    return lView[T_HOST];
+  }
+
+  // Remaining TNode type is `TViewType.Root` which doesn't have a parent TNode.
+  return null;
+}
+
+/**
+ * Calculates the parent TNode and matching LView for a given TNode.
+ *
+ * Determines the node host or walks the declaration tree until it finds
+ * the next valid parent node, if it exists, and sets that.
+ *
+ * @param tNode a `TNode` that we want to find the parent of.
+ * @param lView an `LView` that corresponds to the tNode param.
+ * @param flags an `InjectFlags` that we check to skip @Host.
+ */
+function calculateParentTNodeAndLView(tNode: TNode, lView: LView, flags: InjectFlags): void {
+  ngDevMode && assertTNodeForTView(tNode, lView[TVIEW]);
+  let parentTNode = tNode as TNode | null;
+  let parentLView = lView;
+  let found = false;
+
+  while (!found) {
+    parentTNode = parentTNode!.parent as TNode | null;
+    if (parentTNode === null && !(flags & InjectFlags.Host)) {
+      parentTNode = getDeclarationTNode(parentLView);
+      if (!parentTNode) return;
+
+      // In this case, a parent exists and is definitely an element. So it will definitely
+      // have an existing lView as the declaration view, which is why we can assume it's defined.
+      ngDevMode && assertDefined(parentLView, 'Parent LView should be defined');
+      parentLView = parentLView[DECLARATION_VIEW]!;
+
+      // In Ivy there are Comment nodes that correspond to ngIf and NgFor embedded directives
+      // We want to skip those and look only at Elements and ElementContainers to ensure
+      // we're looking at true parent nodes, and not content or other types.
+      if (parentTNode.type & (TNodeType.Element | TNodeType.ElementContainer)) {
+        found = true;
+      }
+    } else {
+      found = true;
+    }
+  }
+  setCurrentTNodeAndLView(parentTNode, parentLView);
+}
+
+function setCurrentTNodeAndLView(tNode: TNode|null = null, lView: LView|null = null): void {
```

Comment:
It looks like all call sites contain arguments, so we can omit them:
```suggestion
function setCurrentTNodeAndLView(tNode: TNode|null, lView: LView|null): void {
```
I'd also propose to move this function right next to the `currentTNode` and `currentLView` definition for readability (so that it's easier to find these vars + corresponding documentation).

Reply:
This method was removed in the fixup in favor of using the global state object.